src/main/kotlin/io/emeraldpay/dshackle/upstream/ethereum/connectors/EthereumRpcConnector.kt

```diff
@@ -22,28 +23,33 @@ import io.emeraldpay.dshackle.upstream.ethereum.connectors.EthereumConnectorFact
 import io.emeraldpay.dshackle.upstream.forkchoice.AlwaysForkChoice
 import io.emeraldpay.dshackle.upstream.forkchoice.ForkChoice
 import org.slf4j.LoggerFactory
+import reactor.core.publisher.Flux
 import reactor.core.scheduler.Scheduler
 import java.time.Duration
 
 class EthereumRpcConnector(
-    private val connectorType: ConnectorMode,
+    connectorType: ConnectorMode,
     private val directReader: JsonRpcReader,
     wsFactory: EthereumWsConnectionPoolFactory?,
     id: String,
     forkChoice: ForkChoice,
     blockValidator: BlockValidator,
     skipEnhance: Boolean,
     wsConnectionResubscribeScheduler: Scheduler,
-    headScheduler: Scheduler
+    headScheduler: Scheduler,
+    expectedBlockTime: Duration
 ) : EthereumConnector, CachesEnabled {
     private val pool: WsConnectionPool?
     private val head: Head
+    private val liveness: HeadLivenessValidator
 
     companion object {
         private val log = LoggerFactory.getLogger(EthereumRpcConnector::class.java)
     }
 
-    override fun getConnectorMode() = connectorType
+    override fun hasLiveSubscriptionHead(): Flux<Boolean> {
+        return liveness.getFlux() ?: Flux.just(false)
```

Comment:
`Flux.just(false)` can be omitted, as I see `liveness.getFlux()` is never `null`